### ISSUE TITLE
Is there any plan to release training script?

### ISSUE THREAD

**Issue by f1shel** [thu-ml/CRM#3]
Hi, thanks for your wonderful paper and the released inference code!!! I am wondering if there is any plan to release the training script. Your reply will be highly appreciated~

**Comment by SanketDhuri** [comment]
I attempted to develop training for this, but the outcomes aren’t meeting my expectations (my goal was to overfit it for a single object, such as a shoe). Below given repo contains the code for training the model. This code is specifically designed to train 3D geometry, as I am deliberately inputting RGB images (manually captured) and CCM images (previously generated).

Repo [link](https://github.com/SanketDhuri/crm_3d_training.git)

please contribute to it if it has any mistakes / bug found

**Comment by thuwzy** [comment]
Thank you for your interest! The training code for the CRM is quite extensive, and it's difficult to organize it all in a short period. I will start by gradually uploading some of the utility functions I have organized.

**Comment by SanketDhuri** [comment]
> Thank you for your interest! The training code for the CRM is quite extensive, and it's difficult to organize it all in a short period. I will start by gradually uploading some of the utility functions I have organized.

@thuwzy can you take a quick look to the training I have uploaded and plz identify mistakes I am making since I am not getting result as expected

**Comment by thuwzy** [comment]
> > Thank you for your interest! The training code for the CRM is quite extensive, and it's difficult to organize it all in a short period. I will start by gradually uploading some of the utility functions I have organized.
> 
> @thuwzy can you take a quick look to the training I have uploaded and plz identify mistakes I am making since I am not getting result as expected

Hi Sanket, I have read your code and found some problems. (1) the sdf should not be randomly initialized. Instead, it should be got from the output triplanes from the sdfMLP. (2) The learning rate is too high. I suggest 1e-4.
